File: app/src/main/java/com/onemoresecret/crypto/TotpUriTransfer.java
package com.onemoresecret.crypto;

import java.security.InvalidAlgorithmParameterException;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.security.interfaces.RSAPublicKey;

import javax.crypto.BadPaddingException;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.NoSuchPaddingException;

public class TotpUriTransfer extends EncryptedMessage {
    public TotpUriTransfer(byte[] message,
                           RSAPublicKey rsaPublicKey,
                           int rsaTransformationIdx,
                           int aesKeyLength,
                           int aesTransformationIdx) throws
            NoSuchAlgorithmException,
            NoSuchPaddingException,
            IllegalBlockSizeException,
            BadPaddingException,
            InvalidKeyException,
            InvalidAlgorithmParameterException {

        super(message, rsaPublicKey, rsaTransformationIdx, aesKeyLength, aesTransformationIdx);
    }

    @Override
    protected int getApplicationId() {
        return MessageComposer.APPLICATION_TOTP_URI;
    }
}

File: app/src/main/java/com/onemoresecret/crypto/TotpUriTransfer.kt
package com.onemoresecret.crypto


class TotpUriTransfer(
    message: ByteArray,
    rsaPublicKeyMaterial: ByteArray,
    rsaTransformation: RsaTransformation,
    aesKeyLength: Int,
    aesTransformation: AesTransformation
) : EncryptedMessage(
    message,
    rsaPublicKeyMaterial,
    rsaTransformation,
    aesKeyLength,
    aesTransformation
) {
    override val applicationId: Int
        get() = MessageComposer.APPLICATION_TOTP_URI
}
